pkg/config/app/kuma-dp/config.go

```diff
@@ -218,3 +232,40 @@ func (d *ApiServer) Validate() (errs error) {
 	}
 	return
 }
+
+type DNS struct {
+	// If true then builtin DNS functionality is enabled and CoreDNS server is started
+	Enabled bool `yaml:"enabled,omitempty" envconfig:"kuma_dns_enabled"`
+	// CoreDNSPort defines a port that handles DNS requests. When transparent proxy is enabled then iptables will redirect DNS traffic to this port.
+	CoreDNSPort uint32 `yaml:"coreDnsPort,omitempty" envconfig:"kuma_dns_core_dns_port"`
+	// CoreDNSEmptyPort defines a port that always responds with empty NXDOMAIN respond. It is required to implement a fallback to a real DNS
+	CoreDNSEmptyPort uint32 `yaml:"coreDnsEmptyPort,omitempty" envconfig:"kuma_dns_core_dns_empty_port"`
+	// EnvoyDNSPort defines a port that handles Virtual IP resolving by Envoy. CoreDNS should be configured that it first tries to use this DNS resolver and then the real one.
+	EnvoyDNSPort uint32 `yaml:"envoyDnsPort,omitempty" envconfig:"kuma_dns_envoy_dns_port"`
+	// CoreDNSBinaryPath defines a path to CoreDNS binary.
+	CoreDNSBinaryPath string `yaml:"coreDnsBinaryPath,omitempty" envconfig:"kuma_dns_core_dns_binary_path"`
+	// CoreDNSConfigTemplatePath defines a path to a CoreDNS config template.
+	CoreDNSConfigTemplatePath string `yaml:"coreDnsConfigTemplatePath,omitempty" envconfig:"kuma_dns_core_dns_config_template_path"`
+}
+
+func (d *DNS) Sanitize() {
+}
+
+func (d *DNS) Validate() error {
+	if !d.Enabled {
+		return nil
+	}
+	if d.CoreDNSPort > 65353 {
```

Comment:
I wonder if we can use strings and use https://pkg.go.dev/github.com/asaskevich/govalidator#IsPort ?
We have these validations all over the code, so not a request to change it here. Just a food for though.

Reply:
I agree, we need some improvements in config validations. Something similar that we've done for policies.

Here is an issue that is opened for quite some time.
https://github.com/kumahq/kuma/issues/316